torch/csrc/jit/passes/loop_unrolling.cpp

```diff
@@ -227,6 +229,85 @@ void UnrollLoops(Block* block) {
 
 } // anonymous namespace
 
+static void addCondAsOutput(Node* loop) {
+  LoopView loop_view(loop);
+  loop->addInput(loop_view.inputCond());
+  auto block_cond_input = loop_view.bodyBlock()->addInput();
+  block_cond_input->copyMetadata(loop_view.inputCond());
+  auto cond_output_index =
+      loop_view.bodyBlock()->registerOutput(loop_view.nextCond());
+  loop_view.bodyBlock()->outputs()[cond_output_index]->copyMetadata(
+      loop_view.nextCond());
+  auto cond_output = loop->addOutput();
+  cond_output->copyMetadata(loop_view.nextCond());
+}
+
+Node* PeelLoop(Node* n, size_t times) {
+  GRAPH_DEBUG("Peeling the loop ", getHeader(n), " ", times, " times");
+
+  auto graph = n->owningGraph();
+  auto orig_loop = LoopView(n);
+
+  WithInsertPoint wip(n);
+  auto times_const = graph->insertConstant(static_cast<int64_t>(times));
+  // N.B. even though a caller may request to peel `times` iterations
+  // `maxTripCount` of the original loop might be less than that
+  // so we should take the minimum of the two
+  auto min_trip_count =
+      graph->insert(prim::min, {orig_loop.maxTripCount(), times_const});
+
+  // we are going to create a temporary copy of the original loop
```

Comment:
I don't understand this comment. It describes line-for-line the code below. But I don't know why we can't just create the peeled copy, and then modify it.

Reply:
deleted the code